File: backend/src/main/java/com/zk/gymos/entity/Exercise.java
package com.zk.gymos.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

/**
 * A single exercise in the shared library, mapped to {@code public.exercises}.
 * {@link #imageUrl}/{@link #videoUrl} point at media in Supabase Storage.
 */
@Getter
@Setter
@Entity
@Table(name = "exercises")
public class Exercise extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 100)
    private String name;

    /** Target body part, free text (胸 / 背 / 腿 …). */
    @Column(name = "body_part", nullable = false, length = 50)
    private String bodyPart;

    /** Equipment used (杠铃 / 哑铃 / 器械 / 自重 …). */
    @Column(length = 50)
    private String equipment;

    /** 1 (easiest) .. n. */
    @Column
    private Short difficulty = 1;

    @Column(columnDefinition = "text")
    private String description;

    /** Demo image/GIF URL in Supabase Storage. */
    @Column(name = "image_url", columnDefinition = "text")
    private String imageUrl;

    /** Demo video URL in Supabase Storage. */
    @Column(name = "video_url", columnDefinition = "text")
    private String videoUrl;
}

File: app/lib/features/exercise/exercise.dart
/// A single exercise from the backend `/api/exercises`.
class Exercise {
  Exercise({
    required this.id,
    required this.name,
    required this.bodyPart,
    this.equipment,
    this.difficulty,
    this.description,
    this.imageUrl,
    this.videoUrl,
  });

  final String id;
  final String name;
  final String bodyPart;
  final String? equipment;
  final int? difficulty;
  final String? description;
  final String? imageUrl;
  final String? videoUrl;

  factory Exercise.fromJson(Map<String, dynamic> json) {
    return Exercise(
      id: json['id'] as String,
      name: json['name'] as String,
      bodyPart: json['bodyPart'] as String,
      equipment: json['equipment'] as String?,
      difficulty: (json['difficulty'] as num?)?.toInt(),
      description: json['description'] as String?,
      imageUrl: json['imageUrl'] as String?,
      videoUrl: json['videoUrl'] as String?,
    );
  }
}
